UnityProject/Assets/Scripts/Managers/CargoManager.cs

```diff
@@ -0,0 +1,255 @@
+	public List<CargoOrder> CurrentOrders = new List<CargoOrder>();
+	//Cart - current orders, that haven't been payed for/ordered yet
+	public List<CargoOrder> CurrentCart = new List<CargoOrder>();
+
+	public CargoUpdateEvent OnCartUpdate = new CargoUpdateEvent();
+	public CargoUpdateEvent OnShuttleUpdate = new CargoUpdateEvent();
+	public CargoUpdateEvent OnCreditsUpdate = new CargoUpdateEvent();
+	public CargoUpdateEvent OnCategoryUpdate = new CargoUpdateEvent();
+	public CargoUpdateEvent OnTimerUpdate = new CargoUpdateEvent();
+
+	/// <summary>
+	/// Calls the shuttle.
+	/// Server only.
+	/// </summary>
+	public void CallShuttle()
+	{
+		if (!CustomNetworkManager.Instance._isServer)
+		{
+			return;
+		}
+
+		if (CurrentFlyTime > 0f)
+		{
+			return;
+		}
+		if (CustomNetworkManager.Instance._isServer)
+		{
+			CurrentFlyTime = shuttleFlyDuration;
+			//It works so - shuttle stays in centcomDest until timer is done,
+			//then it starts moving to station
+			if (ShuttleStatus == CargoShuttleStatus.DockedCentcom)
+			{
+				ShuttleStatus = CargoShuttleStatus.OnRouteStation;
+				CentcomMessage += "Shuttle is sent back with goods." + "\n";
+				StartCoroutine(Timer(true));
+			}
+			//If we are at station - we start timer and launch shuttle at the same time.
+			//Once shuttle arrives centcomDest - CargoShuttle will wait till timer is done
+			//and will call OnShuttleArrival()
+			else if (ShuttleStatus == CargoShuttleStatus.DockedStation)
+			{
+				CargoShuttle.Instance.MoveToCentcom();
+				ShuttleStatus = CargoShuttleStatus.OnRouteCentcom;
+				CentcomMessage = "";
+				StartCoroutine(Timer(false));
+			}
+		}
+		OnShuttleUpdate?.Invoke();
+	}
+
+	public void LoadData()
+	{
+		Supplies = cargoData.Supplies;
+	}
+
+	private IEnumerator Timer(bool launchToStation)
+	{
+		while (CurrentFlyTime > 0f)
+		{
+			CurrentFlyTime -= 1f;
+			OnTimerUpdate?.Invoke();
+			yield return YieldHelper.Second;
+		}
+
+		if (launchToStation)
+		{
+			CargoShuttle.Instance.MoveToStation();
+		}
+	}
+
+	/// <summary>
+	/// Method is called once shuttle arrives to its destination.
+	/// Server only.
+	/// </summary>
+	public void OnShuttleArrival()
+	{
+		if (!CustomNetworkManager.Instance._isServer)
+		{
+			return;
+		}
+
+		if (ShuttleStatus == CargoShuttleStatus.OnRouteCentcom)
+		{
+			ShuttleStatus = CargoShuttleStatus.DockedCentcom;
+			SpawnOrder();
+		}
+		else if (ShuttleStatus == CargoShuttleStatus.OnRouteStation)
+		{
+			ShuttleStatus = CargoShuttleStatus.DockedStation;
+		}
+		OnShuttleUpdate?.Invoke();
+	}
+
+	public void DestroyItem(ObjectBehaviour item)
+	{
+		//If there is no bounty for the item - we dont destroy it.
+		float credits = CargoManager.Instance.AddCredits(item);
+		if (credits <= 0f)
+		{
+			return;
+		}
+
+		if (CentcomMessage == "")
+			CentcomMessage = "Bounty items recieved.\n";
+		//1 - quantity of items
+		CentcomMessage += "+" + credits + " credits: recieved " + "1" + " " + item.name + ".\n";
```

Comment:
Some grouping would be nice in the future

Reply:
Also use string interpolation for things like this so they're more readable:
![GIF](https://user-images.githubusercontent.com/10403536/58466729-ca594100-8129-11e9-87c4-d93d4b8a01f1.gif)
And there's a typo

Reply:
Aand there's (Clone) text everywhere, maybe use `item.gameObject.ExpensiveName()` here instead